common/lib/xmodule/xmodule/randomize_module.py

```diff
@@ -58,15 +56,24 @@ def __init__(self, *args, **kwargs):
                 else:
                     self.choice = random.randrange(0, num_choices)
 
+        # Now get_children() should return a list with one element
+        self.child_descriptor = self.get_child_descriptor_with_choice
+        self.child = self.get_children
+        if self.child is not None:
+            log.debug("children of randomize module (should be only 1): %s", self.child)
+
+    @property
+    def get_child_descriptor_with_choice(self):
         if self.choice is not None:
-            self.child_descriptor = self.descriptor.get_children()[self.choice]
-            # Now get_children() should return a list with one element
-            log.debug("children of randomize module (should be only 1): %s",
-                      self.get_children())
-            self.child = self.get_children()[0]
-        else:
-            self.child_descriptor = None
-            self.child = None
+            return self.descriptor.get_children()[self.choice]
+        return None
+
+    @property
+    def get_children(self, **kwargs):
```

Comment:
This can't be a property, because it overrides `get_children` (a method on all XBlocks).